### ISSUE TITLE
[LFM] Looking For Maintainer

### ISSUE THREAD

**Issue by Stebalien** [Stebalien/term#93]
So, I thought I'd have time to maintain this but then I got a job (where a large portion of said job is maintaining a bunch of OSS projects...). Unfortunately, that leaves me little time (or motivation) to maintain *this* project.

**Comment by gnzlbg** [comment]
FYI the term crate is now a dependency of Rust's built-in unit testing and benchmarking framework.

**Comment by matteomeli** [comment]
It's also a dependency for rustup: https://github.com/rust-lang/rustup.rs/blob/bb6f65f08ba3e8a6619a033b65611be0f0023e73/src/rustup-cli/download_tracker.rs

**Comment by gnzlbg** [comment]
I've moved libtest to use the termcolor crate, maybe it might make sense to do the same for rustup @matteomeli .

**Comment by matteomeli** [comment]
Yeah that seems reasonable if termcolor is more maintained and has equivalent features. Perhaps open an issue in rustup?

**Comment by derekdreery** [comment]
I'd be interested in maintaining.

**Comment by Stebalien** [comment]
@derekdreery that's awesome! If I fall behind on reviewing PRs, please don't hesitate to bug me (i.e., "be annoying").

**Comment by tarcieri** [comment]
I've filed a (presently merged) informational advisory that this crate is unmaintained and looking for a new maintainer to the RustSec Advisory DB:

https://github.com/RustSec/advisory-db/pull/182

Let us know if you'd prefer we not include it. Thanks!

**Comment by Stebalien** [comment]
Thanks! Note: This crate isn't _exactly_ unmaintained. I'm just not going to be fulfilling feature requests and/or fixing known limitations.

**Comment by sourcefrog** [comment]
> Yeah that seems reasonable if termcolor is more maintained and has equivalent features. Perhaps open an issue in rustup?

For others considering this, I'll just note that `termcolor` doesn't have the cursor control functions like [`carriage_return`](https://stebalien.github.io/doc/term/term/trait.Terminal.html#tymethod.carriage_return) supported by `term`. I'm not sure what is a good cross-platform alternative.

**Comment by tarcieri** [comment]
Perhaps consider crossterm?

https://github.com/crossterm-rs/crossterm

**Comment by tarcieri** [comment]
Perhaps consider crossterm?

https://github.com/crossterm-rs/crossterm

**Comment by thomcc** [comment]
I'm pretty familiar with terminal output, and am aware of most of the relevant issues around it (various issues around terminfo, windows console caveats, etc). I could take over maintenance if that would be desirable.

I normally wouldn't volunteer for something like this for a crate I've never contributed to -- in fact, it feels somewhat weird to do so, but I was convinced to do so by a friend, so here I am.

Some things I'd be interested in adding:

- Fallback to parsing `infocmp` output on platforms that use hashed terminfo databases.
- Handling mintty and other non-console terminal emulators on windows, if possible (right now almost nothing handles this)
- Making the `Terminal` trait more fully featured. Styling, color support beyond 16, more control operations. Still trying to keeping it simple and minimal though.
- Improving ergonomics... It's not really competitive with a lot of the crates in the ecosystem on this front.

This stuff I would just do in a separate crate -- but `term` has users who are in need of a maintainer, I guess. That said, I'm not in the slightest bit offended if you aren't interested.

All that said, I see that https://github.com/Stebalien/term/issues/93#issuecomment-539246132 says it's not actually unmaintained, which makes it feel pretty weird that it has a security advisory out! ... :/

**Comment by Stebalien** [comment]
Well, it's "maintained" in that I'll take a look at bug reports and merge patches. But, other than that, it's mostly frozen (I won't be implementing features).

If you're interested in working on this crate, I'm happy to review patches/proposals and slowly hand off maintainership.
